src/Engines/TemplateEngine.js

```diff
@@ -81,22 +85,22 @@ class TemplateEngine {
     if (this.includesDir) {
       let bench = aggregateBench.get("Searching the file system");
       bench.before();
-      this.extensions.forEach(function (extension) {
+      await Promise.all(this.extensions.map(async function (extension) {
         partialFiles = partialFiles.concat(
-          fastglob.sync(prefix + extension, {
+          await fastglob(prefix + extension, {
             caseSensitiveMatch: false,
             dot: true,
           })
         );
-      });
+      }));
       bench.after();
     }
 
     partialFiles = TemplatePath.addLeadingDotSlashArray(partialFiles);
 
-    for (let j = 0, k = partialFiles.length; j < k; j++) {
+    await Promise.all(partialFiles.map(async (partialFile) => {
```

Comment:
ah, hmm—I’m curious about this `map` usage. You’re Promise.all-ing the return there but the map doesn’t seem to return a promise? Is there some additional efficiency we can get here?

Reply:
By using an async arrow function, the return value will be a Promise.

Reply:
Could probably use `.forEach()` here since I don't see anything that is _returning_ a value within `.map()`. Same w/ L88 above.

Reply:
It's returning `Promise.resolve(undefined)`.
You're _not consuming_ the return value of `Promise.all`.
I'd argue, that `Promise.all` requires an `Array<Promise<any>>` to work on unless you want to experience undefined behaviour.

Reply:
As Ryuno-Ki said, the `.map` call is needed because async functions return a promise, which is awaited by `Promise.all`. `forEach` would cause the function to proceed before the promises are done.

I could rewrite this using `.then` chains instead of async functions so there is a clear return value if preferred.